core/src/main/scala/cats/syntax/unorderedFoldable.scala

```diff
@@ -0,0 +1,79 @@
+package cats
+package syntax
+
+import cats.kernel.CommutativeMonoid
+
+trait UnorderedFoldableSyntax extends UnorderedFoldable.ToUnorderedFoldableOps {
+  implicit final def catsSyntaxUnorderedFoldableOps[F[_]: UnorderedFoldable, A](fa: F[A]): UnorderedFoldableOps[F, A] =
+    new UnorderedFoldableOps[F, A](fa)
+}
+
+final class UnorderedFoldableOps[F[_], A](val fa: F[A]) extends AnyVal {
+  private def orderMaxCommutativeMonoidBy[B](f: A => B)(implicit ev: Order[B]): CommutativeMonoid[Option[A]] =
+    new CommutativeMonoid[Option[A]] {
+      override def empty = None
+      override def combine(optX: Option[A], optY: Option[A]): Option[A] = (optX, optY) match {
+        case (None, y) => y
+        case (x, None) => x
+        case (Some(x), Some(y)) => if (ev.gteqv(f(x), f(y))) Some(x) else Some(y)
```

Comment:
We should be able to prevent some `Some` allocations here by changing this to `case (l @ Some(x), r @ Some(y)) => if (ev.gteqv(f(x), f(y))) l else r`. Similarly for the min monoid below.

Reply:
Yes, you're right.